consensus/src/block_storage/block_tree.rs

```diff
@@ -239,6 +241,72 @@ impl BlockTree {
         self.id_to_quorum_cert.get(block_id).cloned()
     }
 
+    /// Retrieves a Window of Recent Blocks
+    ///
+    /// Returns an [`OrderedBlockWindow`](OrderedBlockWindow) containing the previous `window_size`
+    /// blocks, EXCLUDING the provided `current_block`. Returns an `OrderedBlockWindow` containing
+    /// the recent blocks in ascending order by round (oldest -> newest).
+    ///
+    /// # Parameters
+    /// - `current_block`: The reference block to base the window on.
+    /// - `window_size`: The number of recent blocks to include in the window, excluding the `current_block`.
+    ///
+    /// # Example
+    /// Given a `current_block` with `round: 30` and a `window_size` of 3:
+    ///
+    /// ```text
+    /// get_block_window(current_block, window_size)
+    /// // returns vec![
+    /// //     Block { BlockData { round: 28 } },
+    /// //     Block { BlockData { round: 29 } }
+    /// // ]
+    /// ```
+    ///
+    /// *Note*: The output vector in this example contains 2 blocks, not 3, as only blocks with rounds
+    /// preceding `current_block.round()` are included.
+    pub fn get_ordered_block_window(
+        &self,
+        block: &Block,
+        window_size: Option<u64>,
+    ) -> anyhow::Result<OrderedBlockWindow> {
+        // Block round should never be less than the commit root round
+        ensure!(
+            block.round() >= self.commit_root().round(),
+            "Block round {} is less than the commit root round {}, cannot get_ordered_block_window",
+            block.round(),
+            self.commit_root().round()
+        );
+
+        // window_size is None only if execution pool is turned off
+        let Some(window_size) = window_size else {
+            return Ok(OrderedBlockWindow::empty());
+        };
+        let round = block.round();
+        let window_start_round = calculate_window_start_round(round, window_size);
+        let window_size = round - window_start_round + 1;
```

Comment:
Maybe we can make this a util as well like `calculate_window_size_from_start_round`

Reply:
Hmm, but this is just really simple math. An inclusive range

Reply:
Fair enough, I'm fine either way. I could see it being useful if:
1. We reuse it elsewhere in the code 
2. We include some invariants in the calculation like `window_size > 0`, `window_start_round < round`, etc.